### PR TITLE
LPAL-284 - Cypress tests intermittently fail due to CSRF Mismatch/Session

### PR DESCRIPTION
**Pull request by Raffers**
## Purpose
Adds logging to capture CSRF mismatch & creates a Cloudwatch alarm .

Fixes LPAL-284

## Approach

_Explain how your code addresses the purpose of the change_

## Learning

_Any tips and tricks, blog posts or tools which helped you. Plus anything notable you've discovered about the LPA service_

## Checklist

* [x] I have performed a self-review of my own code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [ ] I have added tests to prove my work
* [ ] The product team have tested these changes

### CODE REVIEW

File: terraform/environment/cloudwatch_alarms.tf [hunk 1]
```diff
@@ -43,3 +43,37 @@ resource "aws_cloudwatch_metric_alarm" "admin_5xx_errors" {
...
 }
+
+resource "aws_cloudwatch_log_metric_filter" "csrf_mistmatch_filter" {
+  name           = "CSRFValuesMismatch"
+  pattern        = "{($.priorityName = \"ERROR\") && ($.message = \"Mismatched CSRF provided\")}"
+  log_group_name = data.aws_cloudwatch_log_group.online-lpa.name
+
+  metric_transformation {
+    name      = "EventCount"
+    namespace = "AWS/ApplicationELB"
```
Comment (williamfalconeruk) [inline terraform/environment/cloudwatch_alarms.tf:54]: perhaps this metric filter namespace - and the one below is not quite right, as it refers to the load balancer. we probably want a custom namespace for this.

File: terraform/environment/cloudwatch_alarms.tf [hunk 2]
```diff
@@ -43,3 +43,37 @@ resource "aws_cloudwatch_metric_alarm" "admin_5xx_errors" {
...
+
+resource "aws_cloudwatch_metric_alarm" "front_csrf_mismatch_errors" {
+  actions_enabled     = true
+  alarm_actions       = [aws_sns_topic.cloudwatch_to_pagerduty.arn]
+  alarm_description   = "CSRF Errors returned to front users for ${local.environment}"
+  alarm_name          = "${local.environment} public front CSRF errors"
+  comparison_operator = "GreaterThanThreshold"
+  datapoints_to_alarm = 2
+  dimensions = {
+    "LoadBalancer" = trimprefix(split(":", aws_lb.front.arn)[5], "loadbalancer/")
```
Comment (williamfalconeruk) [inline terraform/environment/cloudwatch_alarms.tf:67]: possibly refer to the container in this instance as a dimension.

File: terraform/environment/cloudwatch_alarms.tf [hunk 3]
```diff
@@ -43,3 +43,37 @@ resource "aws_cloudwatch_metric_alarm" "admin_5xx_errors" {
...
+  }
+  evaluation_periods        = 2
+  insufficient_data_actions = []
+  metric_name               = "${data.aws_cloudwatch_log_group.online-lpa}:csrf_mistmatch_filter"
+  namespace                 = "AWS/ApplicationELB"
```
Comment (williamfalconeruk) [inline terraform/environment/cloudwatch_alarms.tf:72]: see comment re namespaces above.

File: terraform/environment/cloudwatch_alarms.tf (hunk 2 above)
Comment (williamfalconeruk) [reply terraform/environment/cloudwatch_alarms.tf:67]: or you could omit this.

**Review by charlesmmarshall — APPROVED** [review]

**Pull request merged**